src/backend/gl/src/command.rs

```diff
@@ -215,25 +227,7 @@ impl RawCommandBuffer {
     }
 
     fn push_cmd(&mut self, cmd: Command) {
-        let slice = {
-            let mut memory = self
-                .memory
-                .try_lock()
-                .expect("Trying to record a command buffers, while memory is in-use.");
-
-            let cmd_buffer = match *memory {
-                BufferMemory::Linear(ref mut buffer) => &mut buffer.commands,
-                BufferMemory::Individual { ref mut storage, .. } => {
-                    &mut storage.get_mut(&self.id).unwrap().commands
-                }
-            };
-            cmd_buffer.push(cmd);
-            BufferSlice {
-                offset: cmd_buffer.len() as u32 - 1,
-                size: 1,
-            }
-        };
-        self.buf.append(slice);
+        push_cmd_internal(&self.id, &mut self.memory, &mut self.buf, cmd);
```

Comment:
What's the purpose of this internal function?

Reply:
It's needed to avoid a second mutable borrow of `self` in `begin_renderpass` (see [the commit where it was added](https://github.com/gfx-rs/gfx/pull/1701/commits/72b591d136320c72904812c0ede1092dc6b27b6f)) without copying. Basically it accomplishes a partial borrow of `self`.

I could temporarily copy or collect all of the commands instead (or even use `mem::swap` to move it temporarily or something), but I think we may hit this in a few places. I'm open to more ideas here 😃

Reply:
Having the function is good, but perhaps we could add a short explanation in comments/documentation?